biz.aQute.repository/src/aQute/bnd/repository/fileset/FileSetRepository.java

```diff
@@ -87,11 +92,45 @@ private BridgeRepository getBridge() throws Exception {
 				return null;
 			}
 			ResourceBuilder rb = new ResourceBuilder();
-			try (JarFile jar = new JarFile(file)) {
+			try (Jar jar = new Jar(file)) {
 				Domain manifest = Domain.domain(jar.getManifest());
 				boolean hasIdentity = rb.addManifest(manifest);
 				if (!hasIdentity) {
-					return null;
+					String name;
+					String version;
+					Optional<POM> pom;
+
+					// Check if there's a module name
+					if (jar.getModuleName() != null) {
+						name = jar.getModuleName();
+						version = new MavenVersion(jar.getModuleVersion()).getOSGiVersion()
```

Comment:
This is a bit of a risk since Module Versions, https://docs.oracle.com/javase/9/docs/api/java/lang/module/ModuleDescriptor.Version.html, do not conform to maven version rules or OSGi version rules. So it is possible to take an exception here. We may need to add a ModuleVersion class to deal with the rules of module versions and how we want to map them to OSGi versions.